specs/connector.yaml

```diff
@@ -0,0 +1,573 @@
+    name: id
+    in: path
+    description: |
+      ID of a Callback, stored as a base32 encoded 18 byte identifier.
+    required: true
+    type: string
+    pattern: ^[0-9abcdefghjkmnpqrtuvwxyz]{29}$
+    format: base32ID
+responses:
+  BadRequest:
+    description: Request denied due to invalid request body, path, or headers.
+    schema:
+      $ref: '#/definitions/Error'
+    examples:
+      application/json:
+        type: bad_request
+        message:
+        - Invalid Resource ID Provided
+  Unauthorized:
+    description: Request denied as the provided credentials are no longer valid.
+    schema:
+      $ref: '#/definitions/Error'
+    examples:
+      application/json:
+        type: invalid_grant
+        message:
+        - Provided authorization_code is invalid or does not exist.
+  NotFound:
+    description: Request denied as the requested resource does not exist.
+    schema:
+      $ref: '#/definitions/Error'
+    examples:
+      application/json:
+        type: not_found
+        message:
+        - Resource not found
+  Internal:
+    description: Request failed due to an internal server error.
+    schema:
+      $ref: '#/definitions/Error'
+    examples:
+      application/json:
+        type: internal
+        message:
+        - Internal Server Error
+  TokenBadRequest:
+    description: Request denied due to invalid request body, path, or headers.
+    schema:
+      $ref: '#/definitions/OAuthError'
+    examples:
+      application/json:
+        error: invalid_grant
+        error_description: Provided authorization_code is invalid or does not exist.
+  TokenUnauthorized:
+    description: Request denied as the provided credentials are no longer valid.
+    schema:
+      $ref: '#/definitions/OAuthError'
+    examples:
+      application/json:
+        error: invalid_client
+        error_description: Provided client_id and client_secret do not match
+security:
+- oauth: []
+basePath: /v1
+paths:
+  /credentials/:
+    x-manifold-audience: provider
+    post:
+      summary: Create an OAuth credential pair
+      description: |
+        Create an OAuth 2.0 credential pair for a provider's product.
+        `client_secret` is stored as an `scrypt` hash only; if the value is
+        lost after creation, it cannot be recovered.
+      security:
+        - tokenRequired: []
+      tags:
+        - OAuth
+      parameters:
+        - name: body
+          in: body
+          description: A product id and description for the credential pair.
+          schema: { $ref: '#/definitions/OAuthCredentialCreateRequest' }
+      responses:
+        200:
+          description: The created OAuth 2.0 credential pair.
+          schema: { $ref: '#/definitions/OAuthCredentialCreateResponse' }
+        400: { $ref: '#/responses/BadRequest' }
+        401: { $ref: '#/responses/Unauthorized' }
+        404: { $ref: '#/responses/NotFound' }
+        500: { $ref: '#/responses/Internal' }
+
+  /sso:
+    x-manifold-audience: browser
```

Comment:
we filter the spec files based on annotations like this. you can run `tools/bin/openapi-preprocessor --audience=provider specs/connector.yaml connector.yaml` to get the version suitable for use in this repo.